src/components/REPLInput.tsx

```diff
@@ -0,0 +1,251 @@
+import { Dispatch, SetStateAction, useState } from "react";
+import "../styles/main.css";
+import { ControlledInput } from "./ControlledInput";
+import { commandHandlers } from "./CommandHandlers";
+
+/**
+ * an interface of all the props necessary to run REPLInput and their typings
+ */
+interface REPLInputProps {
+  history: (string | JSX.Element)[];
+  setHistory: Dispatch<SetStateAction<(string | JSX.Element)[]>>;
+  mode: string;
+  setMode: Dispatch<SetStateAction<string>>;
+  setCurrentDataset: Dispatch<SetStateAction<string[][]>>;
+  currentDataset: string[][];
+}
+
+// Mocked datasets
+const mockedDatasets = {
```

Comment:
(super super small) but for readability i found it helpful to put datasets in a different class. also when a developer is switching mocked data with real data later down the line this might be more ideal, to make the switch cleaner?